python/GafferSceneUI/PrimitiveVariableQueryUI.py

```diff
@@ -0,0 +1,515 @@
+			currentPlug != oPlug
+		) :
+
+			suffix = "." + currentPlug.getName() + suffix
+			currentPlug = currentPlug.parent()
+
+	result = prefix + suffix
+
+	return result
+
+##########################################################################
+# Output widget
+##########################################################################
+
+class _OutputWidget( GafferUI.PlugValueWidget ) :
+
+	def __init__( self, childPlug, **kw ) :
+
+		self.__row = GafferUI.ListContainer( GafferUI.ListContainer.Orientation.Vertical, spacing = 4 )
+
+		GafferUI.PlugValueWidget.__init__( self, self.__row, childPlug )
+
+		for i in xrange( 0, 2 ) :
+			self.__row.append(
+				GafferUI.ListContainer( GafferUI.ListContainer.Orientation.Horizontal, spacing = 4 )
+			)
+
+		nameWidget = GafferUI.LabelPlugValueWidget(
+			self.getPlugs(),
+			horizontalAlignment = GafferUI.Label.HorizontalAlignment.Right
+		)
+		nameWidget.label()._qtWidget().setFixedWidth( GafferUI.PlugWidget.labelWidth() )
+		self.__row[0].append(
+			nameWidget,
+			verticalAlignment = GafferUI.Label.VerticalAlignment.Top
+		)
+
+		existsLabelWidget = GafferUI.LabelPlugValueWidget(
+			{ plug["exists"] for plug in self.getPlugs() },
+			horizontalAlignment = GafferUI.Label.HorizontalAlignment.Right
+		)
+		existsLabelWidget.label()._qtWidget().setFixedWidth( 40 )
+		self.__row[0].append(
+			existsLabelWidget,
+			verticalAlignment = GafferUI.Label.VerticalAlignment.Top
+		)
+
+		self.__row[0].append(
+			GafferUI.BoolPlugValueWidget( { plug["exists"] for plug in self.getPlugs() } ),
+			verticalAlignment = GafferUI.Label.VerticalAlignment.Top
+		)
+
+		typeLabelWidget = GafferUI.LabelPlugValueWidget(
+			{ plug['type'] for plug in self.getPlugs() },
+			horizontalAlignment = GafferUI.Label.HorizontalAlignment.Right
+		)
+		typeLabelWidget.label()._qtWidget().setFixedWidth( 30 )
+		self.__row[0].append(
+			typeLabelWidget,
+			verticalAlignment = GafferUI.Label.VerticalAlignment.Top
+		)
+
+		self.__row[0].append(
+			GafferUI.StringPlugValueWidget( { plug["type"] for plug in self.getPlugs() } ),
+			verticalAlignment = GafferUI.Label.VerticalAlignment.Top
+		)
+
+		interpLabelWidget = GafferUI.LabelPlugValueWidget(
+			{ plug["interpolation"] for plug in self.getPlugs() },
+			horizontalAlignment = GafferUI.Label.HorizontalAlignment.Right
+		)
+		interpLabelWidget.label()._qtWidget().setFixedWidth( 80 )
+		self.__row[0].append(
+			interpLabelWidget,
+			verticalAlignment = GafferUI.Label.VerticalAlignment.Top
+		)
+		interpPresetWidget = GafferUI.PresetsPlugValueWidget(
+			{ plug["interpolation"] for plug in self.getPlugs() } )
+		interpPresetWidget.menuButton()._qtWidget().setFixedWidth( 90 )
+		self.__row[0].append( interpPresetWidget )
+
+		self.__row[1].append(
+			GafferUI.Spacer(
+				imath.V2i( GafferUI.PlugWidget.labelWidth(), 0 ),
+				imath.V2i( GafferUI.PlugWidget.labelWidth(), 0 )
+			),
+			verticalAlignment = GafferUI.Label.VerticalAlignment.Top
+		)
+
```

Comment:
Would be great to include a LabelPlugValueWidget for `plug["value"]` here too. That would allow people to drag + drop from the NodeEditor to make new output connections.

Reply:
ok, the reason I did not use a `LabelPlugValueWidget` for `plug["value"]` was that the label part of the widget pushes the value widget to the right so that it no longer aligns with the query value widgets above and also means that with the standard node editor panel width, parts of the editors are not visible (i.e. they extend too far to the right) for example the Colour editors. I did not realise that using a `LabelPlugValueWidget` enabled special drag'n'drop behaviour.